### PR TITLE
add restriction to access login form if logged in

### PR DESCRIPTION
**Pull request by LeeRenJie**
depends on #91

### CODE REVIEW

File: message-me/app/controllers/chatrooms_controller.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 class ChatroomsController < ApplicationController
+  before_action :require_user
```
Comment (tlkiong) [inline message-me/app/controllers/chatrooms_controller.rb:2]: I hate the name `require_user`. Not clear on what this is doing.

File: message-me/app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class SessionsController < ApplicationController
+  before_action :logged_in_redirect, only: [:new, :create]
```
Comment (tlkiong) [inline message-me/app/controllers/sessions_controller.rb:2]:
```suggestion
  before_action :logged_in_redirect, if: :logged_in?, only: [:new, :create]
```

Please test the above. I not sure if my syntax correct

File: message-me/app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -19,4 +21,13 @@ def destroy
...
   end
+
+  private
+
+  def logged_in_redirect
+    if logged_in?
+      flash[:error] = "You are already logged in"
+      redirect_to root_path
+    end
+  end
```
Comment (tlkiong) [inline message-me/app/controllers/sessions_controller.rb:32]:
```suggestion
  def logged_in_redirect
    flash[:error] = "You are already logged in"
    redirect_to root_path
  end
```

**Review by tlkiong — CHANGES_REQUESTED** [review]

File: message-me/app/controllers/sessions_controller.rb (hunk 2 above)
Comment (LeeRenJie) [reply message-me/app/controllers/sessions_controller.rb:2]: yep it works

File: message-me/app/controllers/chatrooms_controller.rb (hunk 1 above)
Comment (LeeRenJie) [reply message-me/app/controllers/chatrooms_controller.rb:2]: I changed it to validate_user_logged_in

**Review by tlkiong — COMMENTED** [review]
Will approve this once https://github.com/LeeRenJie/Learn-Ruby/pull/91 is approved & merged

**Review by tlkiong — APPROVED** [review]

**Pull request merged**